dev/df_new_loader.py

```diff
@@ -34,11 +36,32 @@
 # CREATE DATAFRAME
 df = imed_loader_utils.create_bids_dataframe(loader_params, derivatives)
 
+# COPY DATAFRAME TO TEST get_subdatasets_subjects_list_new FUNCTION (BELOW)
+df2 = df
+
 # DROP "path" AND "parent_path" COLUMNS AND SORT BY FILENAME FOR TESTING PURPOSES WITH data-testing
 df = df.drop(columns=['path', 'parent_path'])
 df = df.sort_values(by=['filename']).reset_index(drop=True)
 
-# SAVE DATAFRAME TO CSV FILE
+# SAVE DATAFRAME TO CSV FILE FOR data-testing
 path_csv = "test_df_new_loader.csv"
 df.to_csv(path_csv, index=False)
 print(df)
+
+# CREATE LOG DIRECTORY
+log_directory = context["log_directory"]
+if not os.path.isdir(log_directory):
+    print('Creating log directory: {}'.format(log_directory))
+    os.makedirs(log_directory)
+else:
+    print('Log directory already exists: {}'.format(log_directory))
+
+# SPLIT TRAIN/VALID/TEST (with "new" functions)
+train_lst, valid_lst, test_lst = imed_loader_utils.get_subdatasets_subjects_list_new(context["split_dataset"],
+                                                                                        df2,
```

Comment:
indentation (should be aligned with the param on the previous line). If you use PyCharm, ctrl+alt+L will fix all indentation and minor PEP8 errors automatically.